retry/get_body_go18_test.go

```diff
@@ -0,0 +1,27 @@
+package http_retry
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"os"
+	"testing"
+)
+
+func Test_getBody_CanRepeatedlyRead(t *testing.T) {
+	rs, err := os.Open("test.txt")
```

Comment:
Could replace this with `bytes.NewReader([]byte("abc"))` for a `io.ReadSeeker` to avoid having to have a file lying about.

Reply:
Good point. I initially started with the way you suggested... It turned out Go HTTP library has special optimisation cases for bytes reader and string reader. So the test would only execise the logic if a file reader is used...

Reply:
(/usr/local/go/src/net/http/request.go:884)